### PR TITLE
[Console] `SymfonyStyle` enhancements

### PR DESCRIPTION
**Pull request by kbond**
| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| Tickets       | n/a
| License       | MIT
| Doc PR        | todo

Adds the following enhancements to `SymfonyStyle`:
1. Add `::progressIterate()`
2. Add `::createTable()` method to reduce duplication and add customization point
3. Use `ConsoleSectionOutput` when creating table (if available) - allows for creating "appendable" tables

### CODE REVIEW

File: src/Symfony/Component/Console/Style/SymfonyStyle.php [hunk 1]
```diff
@@ -421,6 +417,19 @@ public function getErrorStyle()
         return new self($this->input, $this->getErrorOutput());
     }
 
+    public function createTable(): Table
+    {
+        return (new Table($this->output))->setStyle($this->tableStyle());
+    }
+
+    protected function tableStyle(): TableStyle
```
Comment (kbond) [inline src/Symfony/Component/Console/Style/SymfonyStyle.php]: I'm thinking this could be removed and moved into `createTable()` above.

**Comment by chalasr** [comment]
Looks good, thanks. Can you add some test ?

**Comment by kbond** [comment]
Tests added.

**Review by chalasr — APPROVED** [review]

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @kbond.

**Pull request merged**